cdap-common/src/main/java/co/cask/cdap/common/zookeeper/ZooKeeperUtils.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.common.zookeeper;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.Service;
+import org.apache.twill.zookeeper.ZKClientService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * Util class used to refactor ZooKeeper functions.
+ */
+public class ZooKeeperUtils {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ZooKeeperUtils.class);
+
+  private ZooKeeperUtils(){}
+
+  public static void connectWithTimeout(ZKClientService zkClientService, CConfiguration cConf) throws Exception {
```

Comment:
I don't like this util method (and this class). It's too ZK specific. If It's better to have a util class for `Service` (call it `Services`) and have a method like this:

```java
public static void startAndWait(Service service, long timeout, TimeUnit timeoutUnit) throws TimeoutException, InterruptedException, ExecutionException {
  try {
    service.start().get(timeout, timeoutUnit);
  } catch (TimeoutException e) {
    TimeoutException ex = new TimeoutException(String.format("Service '%s' failed to start in %d %s", service, timeout, timeoutUnit));
    ex.addSuppressed(e);
    service.stop();
    throw ex;
  } catch (InterruptedException e) {
    service.stop();
    throw e;
  } catch (ExecutionException e) {
    throw Throwables.propagate(e.getCause());
  }
}
```